src/mjolnir/graphvalidator.cc

```diff
@@ -594,8 +596,11 @@ namespace mjolnir {
         }
       }
 
-      //transit level
-      if (level == TileHierarchy::levels().rbegin()->second.level) {
+      // Check if transit level exists - add transit tiles to queue if so
```

Comment:
the reason this takes long in any case is because it doesnt add tiles that _**do**_ exist in the dir (a small number even if you have planet wide transit), its because it checks for the existance of every possible tile that _**could**_ exist. we should change the loop below to just loop over the files found in the directory, it will certainly be way less and have to do way way way less system calls.